src/js/components/TabForm.js

```diff
@@ -0,0 +1,200 @@
+              {label}
+            </span>
+            <Tooltip content={description} wrapperClassName="tooltip-wrapper
+              media-object-item" wrapText={true} maxWidth={300}
+              scrollContainer=".gm-scroll-view">
+              <i className="icon icon-sprite icon-sprite-mini icon-error" />
+            </Tooltip>
+          </div>
+        </span>
+      </label>
+    );
+  }
+
+  getHeader(title, description) {
+    return (
+      <div key={title} className="form-row-element">
+        <h3 className="form-header flush-bottom">{title}</h3>
+        <p className="flush-bottom">{description}</p>
+      </div>
+    );
+  }
+
+  getSideContent(multipleDefinition) {
+    let {currentTab} = this.state;
+
+    return (
+      <div className="column-mini-12 column-small-4 multiple-form-left-column">
+        <SideTabs
+          onTabClick={this.handleTabClick}
+          selectedTab={currentTab}
+          tabs={Object.values(multipleDefinition)} />
+      </div>
+    );
+  }
+
+  getFormPanels() {
+    let currentTab = this.state.currentTab;
+    let multipleDefinition = this.props.definition;
+    let multipleDefinitionClasses = 'multiple-form-right-column column-mini-12 column-small-8';
+
+    let panels = Object.keys(multipleDefinition).map((formKey, i) => {
+      let panelClassSet = classNames('form', {
+        'hidden': currentTab !== formKey
+      });
+
+      let {definition, description, title} = multipleDefinition[formKey];
+      let formDefinition = [{
+        render: this.getHeader.bind(this, title, description)
+      }].concat(definition);
+
+      return (
+        <div key={i} className="form-panel">
+          <Form
+            className={panelClassSet}
+            formGroupClass="form-group flush"
+            definition={formDefinition}
+            triggerSubmit={this.getTriggerSubmit.bind(this, formKey)}
+            onChange={this.props.onChange}
+            onSubmit={this.handleFormSubmit.bind(this, formKey)} />
+        </div>
+      );
+    });
+
+    // On intial render, we don't want to render with Gemini because it will
+    // cancel the parent's animation, due to it measuring the component.
+    if (!this.state.renderGemini) {
+      return (
+        <div className={multipleDefinitionClasses}>
+          {panels}
+        </div>
+      );
+    }
+
+    return (
+      <GeminiScrollbar
+        autoshow={true}
+        className={multipleDefinitionClasses}
+        ref="geminiForms">
+        {panels}
+      </GeminiScrollbar>
+    );
+  }
+
+  render() {
+    return (
+      <div className={this.props.className}>
+        {this.getSideContent(this.props.definition)}
+        {this.getFormPanels()}
+      </div>
+    );
+  }
+}
+
+TabForm.defaultProps = {
+  className: 'multiple-form row',
+  getTriggerSubmit: function () {},
+  onChange: function () {},
+  onSubmit: function () {}
+};
+
+TabForm.propTypes = {
+  className: React.PropTypes.string,
```

Comment:
Let's have all props listed here. Especially `definition`, which seems to be required for this to work.